File: base/src/main/java/com/listenergao/base/utils/CacheUtils.java
package com.listenergao.base.utils;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

import com.listenergao.base.BaseApplication;
import com.listenergao.base.R;

public class CacheUtils {
    @SuppressLint("StaticFieldLeak")
    private static Context context = BaseApplication.currentApplication();

    private static SharedPreferences SP = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);

    public static void save(String key, String value) {
        SP.edit().putString(key, value).apply();
    }

    public static String get(String key) {
        return SP.getString(key, null);
    }
}

File: base/src/main/java/com/listenergao/base/utils/CacheUtils.kt
package com.listenergao.base.utils

import android.content.Context
import com.listenergao.base.BaseApplication

object CacheUtils {

    private val context = BaseApplication.currentApplication()
    private val SP = context.getSharedPreferences("ListenerGao", Context.MODE_PRIVATE)

    fun save(key: String?, value: String?) {
        SP.edit().putString(key, value).apply()
    }

    fun get(key: String): String? {
        return SP.getString(key, null)
    }
}